### ISSUE TITLE
[IDN]Error parsing certificate request: x509: SAN dNSName is malformed

### ISSUE THREAD

**Issue by wordlesswind** [acmesh-official/acme.sh#3845]
Steps to reproduce
------------------

Debug log
-----------------

```
docker exec acme.sh --issue --dns dns_dp --debug 2  \
  --server letsencrypt_test  \
  --keylength ec-384 --ocsp \
  -d 清靈語.中国 -d *.清靈語.中国  \
  -d 清靈語.中國 -d *.清靈語.中國  \
  -d 清灵语.中国 -d *.清灵语.中国  \
  -d 清灵语.中國 -d *.清灵语.中國
```
[acme.sh.log](https://github.com/acmesh-official/acme.sh/files/7667404/acme.sh.log)

**Comment by vitaly80** [comment]
Same!

` Sign failed, finalize code is not 200.
[Sun Nov 21 12:40:34 MSK 2021] {
  "type": "urn:ietf:params:acme:error:malformed",
  "detail": "Error parsing certificate request: x509: SAN dNSName is malformed",
  "status": 400
}
`

**Comment by Neilpang** [comment]
fixed,  please try with the latest dev branch.

```
acme.sh  --upgrade  -b dev
```

**Comment by wordlesswind** [comment]
> --debug 2

非常感谢，不过似乎还是遇到了问题：`idn: could not convert from ANSI_X3.4-1968 to UTF-8`
```
acme.sh --issue --dns dns_dp  \
  --server letsencrypt_test  \
  -k ec-384 --ocsp --debug 2  \
  -d 清靈語.中国 -d *.清靈語.中国
```
[acme.sh_dev.log](https://github.com/acmesh-official/acme.sh/files/7833210/acme.sh_dev.log)

**Comment by Neilpang** [comment]
你这个域名在我这没有问题.

貌似是因为你的terminal encoding 问题.
https://stackoverflow.com/questions/48743106/whats-ansi-x3-4-1968-encoding


在你的 terminal 尝试这个命令:

```
idn  清靈語.中国
```

看看能成功不.

**Comment by wordlesswind** [comment]
> 你这个域名在我这没有问题.
> 
> 貌似是因为你的terminal encoding 问题. https://stackoverflow.com/questions/48743106/whats-ansi-x3-4-1968-encoding
> 
> 在你的 terminal 尝试这个命令:
> 
> ```
> idn  清靈語.中国
> ```
> 
> 看看能成功不.

噢是我犯病了，我是在初始的 debian 容器里测试的，想拿容器当即丢的虚拟机来用，忘了配置环境（

**Comment by wordlesswind** [comment]
fixed in e67d26c

**Comment by vitaly80** [comment]
Thanks, but all of my domains going to idn(
